hazelcast-jet-s3/src/main/java/com/hazelcast/jet/s3/S3Sinks.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.s3;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.hazelcast.jet.function.FunctionEx;
+import com.hazelcast.jet.function.SupplierEx;
+import com.hazelcast.jet.pipeline.Sink;
+import com.hazelcast.jet.pipeline.SinkBuilder;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Contains factory methods for creating AWS S3 sinks.
+ */
+public final class S3Sinks {
+
+    private static final int DEFAULT_BATCH_SIZE = 1024;
+
+    private S3Sinks() {
+    }
+
+    /**
+     * Convenience for {@link #s3(String, int, SupplierEx, FunctionEx)}.
+     * Creates an S3 client with given credentials and region. Uses {@link
+     * Object#toString()} for {@code toStringFn}.
+     */
+    @Nonnull
+    public static <T> Sink<? super T> s3(
+            @Nonnull String bucketName,
+            @Nonnull String accessKeyId,
```

Comment:
I think it would be better to group accessKeyId / secret / region into an object?

Reply:
like `S3Credentials` or `S3Parameter` ?

Reply:
Yeah S3Credentials is ok I think. They can also be defined by environment variables? Are these optional then?

Reply:
what about the `AWSCredentials` object thats comes in the API